File: problems/LeetCode/7.Reverse-Integer.js
/**
 * @param {Integer}
 * @return {Integer}
 */
function reverseInt(int) {
  let negative = false;
  let reversed = 0;
  if (int < 0) {
    negative = true;
    int *= -1;
  }

  while (int > 0) {
    let number = int % 10;
    reversed = reversed * 10 + number;
    int = Math.floor(int / 10);
  }

  if (reversed > Number.MAX_SAFE_INTEGER) {
    return 0;
  }

  return negative ? -reversed : reversed;
}

console.log(reverseInt(-1534236469));

File: problems/LeetCode/7.Reverse-Integer.py

# @type number: int
# @return: int
def reverse_integer(number):

  if number < -2**31 or number > 2**31 - 1:
    return 0

  negative_number = False
  reversed = 0
  if number < 0:
    negative_number = True
    number = number * -1
  
  while number:
    temp = number % 10
    reversed = reversed * 10 + temp
    number = int(number / 10)

  return (-reversed if negative_number else reversed)


print(reverse_integer(2**31))

print (2**31-1)